### DISCUSSION TITLE
[Blender + AMD] Whole system freezing after trying to select HIP as Cycles device

### DISCUSSION THREAD

**Discussion by TheRedstoneDEV-DE** [rocm-arch/rocm-arch#978] [Support]
Hello,
I have a problem, since I switched from a NVidia RTX 3060 to an AMD **RX 6750 XT** two weeks ago or so, my whole system is locking up, just when trying to select HIP as render device. I don't even get to see the List of availlable devices in 
Blender 4.0.2, before my system totally locks up.

My system specs look like this:
`inxi -b`:
```
System:
  Host: schlingel Kernel: 6.6.7-arch1-1 arch: x86_64 bits: 64
    Desktop: KDE Plasma v: 5.27.10 Distro: Arch Linux
Machine:
  Type: Desktop Mobo: Micro-Star model: B450-A PRO MAX (MS-7B86) v: 4.0
    serial: <superuser required> UEFI: American Megatrends LLC. v: M.F0
    date: 03/22/2022
CPU:
  Info: 8-core AMD Ryzen 7 5700X [MT MCP] speed (MHz): avg: 3243
    min/max: 2200/4662
Graphics:
  Device-1: AMD Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT]
    driver: amdgpu v: kernel
  Display: wayland server: X.org v: 1.21.1.10 with: Xwayland v: 23.2.3
    compositor: kwin_wayland driver: X: loaded: modesetting dri: radeonsi
    gpu: amdgpu resolution: 3440x1440
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 23.2.1-arch1.2
    renderer: AMD Radeon RX 6750 XT (navi22 LLVM 16.0.6 DRM 3.54 6.6.7-arch1-1)
Network:
  Device-1: Intel Ethernet I225-V driver: igc
  Device-2: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet
    driver: r8169
Drives:
  Local Storage: total: 2.49 TiB used: 1.18 TiB (47.5%)
Info:
  Processes: 382 Uptime: 4h 35m Memory: total: 32 GiB available: 31.27 GiB
  used: 4.74 GiB (15.2%) Shell: Bash inxi: 3.3.31
```
 (I'm also running Wayland if that matters)
 
And my kernel log looks like this after the freeze:
`journalctl -r`:
https://pastebin.com/yjj4P1TV 
or 
[kernel.log](https://github.com/rocm-arch/rocm-arch/files/13708028/kernel.log)

Just a sidenote, I tried the same thing on my Laptop which is also running Arch Linux 
and has an AMD Ryzen 5 5500U inside and got the same issue, so it shouldn't be a problem 
with a not completely purged NVidia software or something like this (I'm sure I uninstalled 
everything NVidia related properly).

I'm looking forward to a reply. 

However if this issue somehow should be duplicate of another Issue, 
just linke me there and point me in the right direction, the only similar 
I've seen was the one on a steam deck, which was closed due to a not 
officially supported GPU.

Kind regards,
TheRedstoneDEV-DE.

**Comment by swarminglogic** [comment]
Seems to be an issue with ROCm 5.7.x and unfortunately also 6.0.0 (so far). See thread here: https://github.com/ROCm/HIP/issues/3353

A temporary workaround might to be to revert to 5.6.x which can be done somewhat easily by installing this (which conflicts with most ROCm based packages currently at 5.7): https://aur.archlinux.org/packages/hip-runtime-amd-blender

For posterity, once  https://github.com/ROCm/HIP/issues/3353 is resolved, the packages in conflict I had were:

- hip-runtime-amd
- comgr
- hsa-rocr
- hsakmt-roct
- rocm-device-libs
- rocm-llvm
- rocminfo

I will update this once I know how it went.

Edit: Seems to work OK. Had to reselect the GPU in the Settings window, otherwise it ended up defaulting to use the CPU, which was strange. Also, rocm-smi utility incorrectly reports GPU% which ends up being stuck at 100% for quite some time, or until I closed blender. Power consumption and temperature (and the lack of doing anything) suggested that shouldn't be the case. Not sure what's going on there.

Cycles performance was great (for AMD GPU)

PS: Even though the system ROCm was reverted to 5.6.x, stable diffusion still worked fine on 5.7.x on the PyTorch development branch.

**Comment by acxz** [comment]
Closing this discussion since `blender` and its dependent `hip`/`rocm` packages are packaged in `[extra]` are not part of the packages that this repo maintains. Please open up bug reports on the issue page of the `blender` package here: https://gitlab.archlinux.org/archlinux/packaging/packages/blender/-/issues
